### PR TITLE
Dynamic helper text for Authorizing Official

### PR DESCRIPTION
**Pull request by igorkorenfeld**
## 🗣 Description ##

This PR adds content for the authority official for each type of government organization, and and implements logic to display the relevant text based on the registrant's organization type. 

## 💭 Motivation and context ##
Part of #330 . If this approach and implementation looks good, I will extend it to the .gov page as well.

### CODE REVIEW

File: src/registrar/templates/application_authorizing_official.html [hunk 1]
```diff
@@ -12,8 +12,30 @@ <h2>Who is the authorizing official for your organization?</h2>
   <p>Your authorizing official is the person within your organization who can authorize your domain request. This is generally the highest ranking or highest elected official in your organization. Read more about <a href="{% url 'todo' %}">who can serve as an authorizing official</a>.
   </p>
 
-  <div class="ao-example">
-    {% include "includes/ao_example__city.html" %}
+  <div class="ao_example">
+    {% if is_federal %}
+      {% if federal_type == 'executive' %}
```
Comment (SSPJ) [inline src/registrar/templates/application_authorizing_official.html:17]: For code organization, I might have favored putting this `if` block in an include and all of the example text inside of it over multiple smaller files. That's a matter of preference and this is fine.

File: src/registrar/templates/includes/ao_example__exec.html [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<p>Domain requests from executive branch agencies must be authorized by <strong>CIOs</strong> or <strong>agency heads</strong>.</p>
```
Comment (SSPJ) [inline src/registrar/templates/includes/ao_example__exec.html:1]: I would very much prefer that all abbreviations be [eliminated or at least defined on first usage](https://www.plainlanguage.gov/guidelines/words/minimize-abbreviations/) (within a page):

`<strong>Chief Information Officers</strong>` or 

`<strong>Chief Information Officers (<abbr>CIO</abbr>s)</strong>`

CC @michelle-rago

File: src/registrar/tests/test_views.py [hunk 3]
```diff
@@ -740,6 +740,70 @@ def test_application_type_of_work_interstate(self):
 
         self.assertContains(contact_page, self.TITLES[Step.TYPE_OF_WORK])
 
+    def test_application_ao_dynamic_text(self):
+        type_page = self.app.get(reverse("application:")).follow()
```
Comment (SSPJ) [inline src/registrar/tests/test_views.py:744]: Kudos for inclusion of tests 💯

**Review by SSPJ — APPROVED** [review]
LGTM, @igorkorenfeld ! 👍 

The Basics
- [x] I read the code
- [x] I ran the code

Accessibility
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [x] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - ❌  Form instructions are associated with inputs
    We are missing some `aria-describedby` attributes; but these are out of scope of this PR (your code does not introduce any new issues, I'm noting an existing issue for awareness).
    - [x] Headings are nested properly. Heading elements are nested in a logical way
    - [x] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’

File: src/registrar/views/application.py [hunk 4]
```diff
@@ -365,6 +365,13 @@ class AuthorizingOfficial(ApplicationWizard):
     template_name = "application_authorizing_official.html"
     forms = [forms.AuthorizingOfficialForm]
 
+    def get_context_data(self):
+        context = super().get_context_data()
+        context["is_federal"] = self.application.is_federal()
```
Comment (neilmb) [inline src/registrar/views/application.py:370]: **thought (non-blocking)**: This is the second form  that needed `context["is_federal"]`. Maybe that is common enough that we could move it into the base class and provide that context in all of the forms? The change doesn't need to be made here, just noticing that we reached the second gate in the "Rule of Three".

Comment (igorkorenfeld) [reply src/registrar/views/application.py:370]: Ah good point, and we will likely reach the 3rd gate when doing the .gov domain examples. I can change it as part of that work.

**Pull request merged**